src/modules/hip/kernel/tensor_stddev.hpp

```diff
@@ -0,0 +1,501 @@
+    }
+}
+
+// -------------------- Set 0 - Reduction Stage 2 --------------------
+
+template <typename T>
+__global__ void tensor_stddev_grid_result_hip(T *inputSrcPtr,
+                                              float *srcPtr,
+                                              uint xBufferLength,
+                                              float *dstPtr,
+                                              RpptROIPtr roiTensorPtrSrc)
+{
+    int id_x = hipThreadIdx_x * 8;
+    int id_z = hipBlockIdx_z;
+
+    __shared__ float partialVariance_smem[1024];                    // 8192 floats of src reduced to 1024 in a 1024 x 1 thread block
+    partialVariance_smem[hipThreadIdx_x] = 0.0f;                    // initialization of _smem to 0 using all 1024 x 1 threads
+
+    if (id_x >= xBufferLength)
+        return;
+
+    int xDiff = xBufferLength - (xBufferLength & ~7);               // difference between roiWidth and alignedLength, where alignedLength = roiWidth & ~7
+    uint srcIdx = (id_z * xBufferLength) + id_x;
+
+    d_float8 src_f8;
+    rpp_hip_load8_and_unpack_to_float8(srcPtr + srcIdx, &src_f8);   // load 8 pixels to local memory
+    if (id_x + 8 > xBufferLength)
+        for(int i = xDiff; i < 8; i++)
+            src_f8.f1[i] = 0.0f;                                    // local memory reset of invalid values (from the vectorized global load) to 0.0f
+    src_f8.f4[0] += src_f8.f4[1];                                   // perform small work of vectorized float4 addition
+    partialVariance_smem[hipThreadIdx_x] += (src_f8.f1[0] +
+                                             src_f8.f1[1] +
+                                             src_f8.f1[2] +
+                                             src_f8.f1[3]);         // perform small work of reducing float4s to float using 1024 x 1 threads and store in _smem
+    __syncthreads();                                                // syncthreads after _smem load
+
+    // Reduction of 1024 floats on 1024 threads per block in x dimension
+    for (int threadMax = 512; threadMax >= 1; threadMax /= 2)
+    {
+        if (hipThreadIdx_x < threadMax)
+            partialVariance_smem[hipThreadIdx_x] += partialVariance_smem[hipThreadIdx_x + threadMax];
+        __syncthreads();
+    }
+
+    // Final store to dst
+    if (hipThreadIdx_x == 0)
+    {
+        int totalElements = roiTensorPtrSrc[id_z].xywhROI.roiHeight * roiTensorPtrSrc[id_z].xywhROI.roiWidth;
+        stddev_hip_compute(inputSrcPtr, sqrt(partialVariance_smem[0] / totalElements), &dstPtr[id_z]);
+    }
+}
+
+template <typename T>
+__global__ void tensor_stddev_grid_3channel_result_hip(T *inputSrcPtr,
+                                                       float *srcPtr,
+                                                       uint xBufferLength,
+                                                       float *dstPtr,
+                                                       RpptROIPtr roiTensorPtrSrc)
+{
+    int id_x = hipThreadIdx_x;
```

Comment:
Why no vectorization?

Reply:
Now we need to load and perform addition for all 4 values of partialValuesPerBlock
Using rpp_hip_load24_pkd3_and_unpack_to_float24_pln3 wont be compatible with this since there are 4 values per block instead of 3 values
using float4 variable of d_float24 also make it difficult to interpret and reset values if id_x + 8 > bufferLength

So used float4 load here